src/browser/modules/Stream/FrameContainer.tsx

```diff
@@ -0,0 +1,166 @@
+  'cypher-script': CypherScriptFrame,
+  'user-list': UserList,
+  'user-add': UserAdd,
+  'change-password': ChangePasswordFrame,
+  pre: PreFrame,
+  play: PlayFrame,
+  'play-remote': PlayFrame,
+  history: HistoryFrame,
+  param: ParamsFrame,
+  params: ParamsFrame,
+  connection: ConnectionFrame,
+  disconnect: DisconnectFrame,
+  schema: SchemaFrame,
+  help: HelpFrame,
+  queries: QueriesFrame,
+  sysinfo: SysInfoFrame,
+  status: ServerStatusFrame,
+  'switch-success': ServerSwitchFrame,
+  'switch-fail': ServerSwitchFrame,
+  'use-db': UseDbFrame,
+  'reset-db': UseDbFrame,
+  dbs: DbsFrame,
+  style: StyleFrame,
+  default: DefaultFrame
+}
+
+const getFrameComponent = (frameData: FrameStack): React.ComponentType<any> => {
+  const { cmd, type } = frameData.stack[0]
+  let MyFrame = nameToFrame[type] || nameToFrame.default
+
+  if (type === 'error') {
+    try {
+      const command = cmd.replace(/^:/, '')
+      const Frame = command[0].toUpperCase() + command.slice(1) + 'Frame'
+      MyFrame = require('./Extras/index')[Frame] || nameToFrame['error']
+    } catch (e) {}
+  }
+  return MyFrame
+}
+
+type FrameContainerProps = {
+  frameData: FrameStack
+  activeConnectionData: Connection | null
+}
+
+export function FrameContainer(props: FrameContainerProps): JSX.Element {
+  const {
+    isFullscreen,
+    toggleFullscreen,
+    isCollapsed,
+    toggleCollapse
+  } = useSizeToggles()
+  const frame = props.frameData.stack[0]
+  const [exportItems, setExportItems] = useState<ExportItem[]>([])
+  const frameProps: BaseFrameProps = {
+    frame: { ...frame, isPinned: props.frameData.isPinned },
+    activeConnectionData: props.activeConnectionData,
+    stack: props.frameData.stack,
+    isFullscreen,
+    isCollapsed,
+    setExportItems
+  }
+  const FrameComponent = getFrameComponent(props.frameData)
+
+  return (
+    <StyledFrame
+      className={isFullscreen ? 'is-fullscreen' : ''}
+      data-testid="frame"
+      fullscreen={isFullscreen}
+    >
+      <FrameTitlebar
+        frame={frame}
+        pinned={props.frameData.isPinned}
+        fullscreen={isFullscreen}
+        fullscreenToggle={toggleFullscreen}
+        collapse={isCollapsed}
+        collapseToggle={toggleCollapse}
+        togglePin={() => undefined}
+      />
+      <ContentContainer>
+        <FrameEditor
+          frame={frame}
+          fullscreenToggle={toggleFullscreen}
+          exportItems={exportItems}
+        />
+        <FrameComponent {...frameProps} />
+      </ContentContainer>
+    </StyledFrame>
+  )
+}
+
+function useSizeToggles() {
+  const [isCollapsed, setCollapsed] = useState(false)
+  const [isFullscreen, setFullscreen] = useState(false)
+
+  function toggleCollapse() {
+    setCollapsed(coll => !coll)
+    setFullscreen(false)
+  }
+
+  function toggleFullscreen() {
+    setFullscreen(full => !full)
+    setCollapsed(false)
+  }
+
+  return {
+    isCollapsed,
+    isFullscreen,
+    toggleCollapse,
+    toggleFullscreen
+  }
```

Comment:
This is a hook. I noticed there's a `src/browser/hooks` folder contains several hooks, and some hooks files inside of components files or folders. Better to consider the consistency of file structure and do some refactor.
This is not related to this PR. Just write it down so we don't forget it.

Reply:
I disagree that we should move all hooks there, I think we should only move those that are used at multiple places to the general folder. Similar to how we only move reusable components to the root, and keep those only used locally close to where it is used.